src/Runner.Worker/Handlers/CompositeActionOutputHandler.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using GitHub.DistributedTask.ObjectTemplating.Tokens;
+using GitHub.DistributedTask.Pipelines.ContextData;
+using GitHub.DistributedTask.WebApi;
+using GitHub.Runner.Common;
+using GitHub.Runner.Sdk;
+using Pipelines = GitHub.DistributedTask.Pipelines;
+
+namespace GitHub.Runner.Worker.Handlers
+{
+    [ServiceLocator(Default = typeof(CompositeActionOutputHandler))]
+    public interface ICompositeActionOutputHandler : IHandler
+    {
+        CompositeActionExecutionData Data { get; set; }
+    }
+
+    public sealed class CompositeActionOutputHandler : Handler, ICompositeActionOutputHandler
+    {
+        public CompositeActionExecutionData Data { get; set; }
+        public Task RunAsync(ActionRunStage stage)
+        {
+            // Evaluate the mapped outputs value
+            if (Data.Outputs != null)
+            {
+                // Evaluate the outputs in the steps context to easily retrieve the values
+                var evaluator = ExecutionContext.ToPipelineTemplateEvaluator();
+
+                DictionaryContextData actionOutputs = evaluator.EvaluateCompositeOutputs(Data.Outputs, ExecutionContext.ExpressionValues, ExecutionContext.ExpressionFunctions);
```

Comment:
Because of the back compat thing... Be careful to handle the structure, and only add the "value"

```
time:
  description: asdf
  value: the value
```